extensions/amp-analytics/0.1/events.js

```diff
@@ -306,6 +334,93 @@ export class EventTracker {
   add(unusedContext, unusedEventType, unusedConfig, unusedListener) {}
 }
 
+/**
+ * Tracks custom browser events.
+ */
+export class CustomBrowserEventTracker extends EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+
+    /** @private {?Observable<!Event>} */
+    this.observables_ = new Observable();
+
+    /** @private {?Object<BrowserEventType:Boolean>} */
+    this.hasListener_ = new Object(null);
+
+    /** @private {?function(!Event)} */
+    this.boundOnSession_ = this.observables_.fire.bind(this.observables_);
+
+    /** @private {function(!Event):void} */
+    this.debouncedBoundOnSession_ = debounce(
+      this.root.ampdoc.win,
+      this.boundOnSession_,
+      SESSION_DEBOUNCE_TIME_MS
+    );
+  }
+
+  /** @override */
+  dispose() {
+    const root = this.root.getRoot();
+    Object.keys(this.hasListener_).forEach((eventName) => {
+      root.removeEventListener(eventName, this.debouncedBoundOnSession_);
+    });
+    this.boundOnSession_ = null;
+    this.observables_ = null;
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    userAssert(
+      isExperimentOn(this.root.ampdoc.win, 'analytics-browser-events'),
+      'expected global "analytics-browser-events" experiment to be enabled'
+    );
+
+    const {
+      'on': eventName,
+      'selectionMethod': selectionMethod = null,
+      'selector': selector,
+    } = config;
+
+    userAssert(
+      selector.length,
+      'Missing required selector on browser event trigger'
+    );
+    assertUniqueSelectors(selector);
+
+    const targetPromises = this.root.getElements(
+      context,
+      selector,
+      selectionMethod,
+      false
+    );
+
+    if (!this.hasListener_[eventName]) {
+      this.root
+        .getRootElement()
+        .addEventListener(eventName, this.debouncedBoundOnSession_, true);
+      this.hasListener_[eventName] = true;
+    }
+
+    return this.observables_.add((event) => {
+      if (event.type !== eventName) {
+        return;
+      }
+      targetPromises.then((targets) => {
+        targets.forEach((target) => {
+          const el = event.target;
+          if (!target.contains(el)) {
+            return;
+          }
+          listener(new AnalyticsEvent(target, eventName, event.detail));
```

Comment:
`event.detail` could potentially leak user information, let's set it to an empty object and add a TODO.

```suggestion
          // TODO(kalemuw): Allowlist properties from event.detail to pass as vars.
          listener(new AnalyticsEvent(target, eventName, {}));
```